addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -573,193 +675,109 @@ export class OdooEditor extends EventTarget {
                 const node = this.idFind(record.id);
                 if (node) {
                     node.setAttribute(record.attributeName, record.value);
+                    DOMPurify.sanitize(node, { IN_PLACE: true });
                 }
```

Comment:
Does `idFind` recover the *live* node (the one in the document)? Because in that case it seems like we first inject an unvalidated attribute then validate it, which is way too late no?

Reply:
From what I tested, it seems that the code does not execute as soon as the attribute is set. It is done on a subsequent tick so nothing is called.
But to be more secure I have changed it to:
```js
                    const parent = node.parentNode;
                    const next = node.nextSibling;
                    node.remove();
                    node.setAttribute(record.attributeName, record.value);
                    DOMPurify.sanitize(node, { IN_PLACE: true });
                    if (next) {
                        next.before(node);
                    } else {
                        parent.append(node);
                    }
```